libraries/Microsoft.Bot.Builder.Dialogs/Waterfall.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// Dialog optimized for prompting a user with a series of questions. Waterfalls accept a stack of
+    /// functions which will be executed in sequence.Each waterfall step can ask a question of the user
+    /// and the users response will be passed as an argument to the next waterfall step.
+    /// </summary>
+    public class Waterfall : Dialog
```

Comment:
>Waterfall [](start = 17, length = 9)

WaterfallDialog ?

Reply:
Sure... That's a change.  It's almost completely hidden from developers on the JS side so I don't care too much about how wordy the name is.